### PR TITLE
Rust bindings refactoring and bip39_mnemonic impl

### PR DESCRIPTION
**Pull request by fckt**
~~there is a problem with a Java build~~

### CODE REVIEW

File: rust/signer/src/android.rs [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+    sys_env: *mut jni::sys::JNIEnv,
+  }
+
+  impl TestVM {
+    fn env<'a>(&'a self) -> jni::JNIEnv<'a> {
+      jni::JNIEnv::from(self.sys_env)
+    }
+  }
+
+  unsafe fn test_vm() -> TestVM {
```
Comment (fckt) [inline rust/signer/src/android.rs]: does not work ATM

**Comment by fckt** [comment]
Can be merged. 
The UI impl. for `bip39 mnemonic` will be delivered as a separate PR.

**Comment by fckt** [comment]
It's worth to mention that it would be much smoother to make [changes](https://github.com/paritytech/parity-signer/compare/mh-bip39-rs25519#diff-eedc50c68d2783d5d5a7ea8f6646824d) to `lib.rs` just after merge of this PR.

**Comment by maciejhirsz** [comment]
Closing in favor of #234.

**Pull request closed without merging**